Cargo.lock

```diff
@@ -4140,13 +4131,13 @@ dependencies = [
 [[package]]
 name = "reedline"
 version = "0.21.0"
-source = "git+https://github.com/nushell/reedline.git?branch=main#cf841beb92affc7cb5cdc16bddefdebcd001f8c9"
+source = "git+https://github.com/nushell/reedline.git?branch=main#f15f0fb413f57fa9da86d656cf8c2eb761b3cfdf"
 dependencies = [
  "chrono",
  "crossterm",
  "fd-lock",
  "itertools",
- "nu-ansi-term 0.47.0",
+ "nu-ansi-term",
```

Comment:
Yes, while a `[patch.crates-io]` is active for reedline, the `Cargo.lock` matters more. That's the result of running `cargo update -p reedline`

Reply:
ooooh i see

this is because `reedline` is a Git dependency in the top-level `Cargo.toml`, right?

Reply:
Because the whole workspace has a `[patch]` override all workspace crates will replace with that git repo branch. we don't specify a particular commit so a `cargo update` can pull in the latest changes.

Recommended reading for any rust crate maintenance work: https://doc.rust-lang.org/cargo/reference/index.html

Reply:
interesting, thanks for the quick answer, this makes sense now :relieved: